From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 20 Mar 2026 19:16:37 +0100
Subject: notifications: Try to use the portal ID for an application if
 defined

For both flatpak and snap applications we should try to use their portal
ID to configure the notifications, rather than their desktop ID

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3353
---
 panels/notifications/cc-notifications-panel.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/panels/notifications/cc-notifications-panel.c b/panels/notifications/cc-notifications-panel.c
index 087151f..6b3d6dc 100644
--- a/panels/notifications/cc-notifications-panel.c
+++ b/panels/notifications/cc-notifications-panel.c
@@ -457,7 +457,10 @@ select_app (CcNotificationsPanel *self,
 
   app = g_object_get_qdata (G_OBJECT (row), application_quark ());
 
-  app_id = cc_util_app_get_app_id (app->app_info);
+  app_id = cc_util_app_get_portal_id (app->app_info);
+
+  if (!app_id)
+    app_id = cc_util_app_get_app_id (app->app_info);
 
   page = cc_app_notifications_page_new (app_id, g_app_info_get_name (app->app_info), app->settings, self->master_settings, self->perm_store);
   cc_panel_push_subpage (CC_PANEL (self), ADW_NAVIGATION_PAGE (page));
